samples/dotnet/EdgeX509AuthDownstreamDevice/Program.cs

```diff
@@ -0,0 +1,300 @@
+                if (certObject is Org.BouncyCastle.X509.X509Certificate x509Cert)
+                {
+                    chain.Add(new X509CertificateEntry(x509Cert));
+                }
+                // when processing certificates generated via openssl certObject type is of AsymmetricCipherKeyPair
+                if (certObject is AsymmetricCipherKeyPair)
+                {
+                    certObject = ((AsymmetricCipherKeyPair)certObject).Private;
+                }
+                if (certObject is RsaPrivateCrtKeyParameters)
+                {
+                    keyParams = ((RsaPrivateCrtKeyParameters)certObject);
+                }
+
+                certObject = pemReader.ReadObject();
+            }
+
+            if (keyParams == null)
+            {
+                throw new InvalidOperationException("Private key is required");
+            }
+
+            store.SetKeyEntry("AzureIoTClient", new AsymmetricKeyEntry(keyParams), chain.ToArray());
+            using (var p12File = new MemoryStream())
+            {
+                store.Save(p12File, new char[] { }, new SecureRandom());
+
+                var cert = new X509Certificate2(p12File.ToArray());
+                return (cert, certsChain);
+            }
+        }
+
+
+        /// <summary>
+        /// First install any CA certificate provided by the user to connect to the Edge device.
+        /// Next attempt to connect to the Edge device and send it MESSAGE_COUNT
+        /// number of telemetry data messages.
+        ///
+        /// Note: Either set the MESSAGE_COUNT environment variable with the number of
+        /// messages to be sent to the IoT Edge runtime or set it in the launchSettings.json.
+        /// </summary>
+        static void Main()
+        {
+            if (string.IsNullOrEmpty(iothubHostname))
+            {
+                throw new ArgumentException("IoT Hub hostname cannot be null or empty.");
+            }
+
+            if (string.IsNullOrEmpty(downstreamDeviceId))
+            {
+                throw new ArgumentException("Downstream device id cannot be null or empty.");
+            }
+
+            if (string.IsNullOrEmpty(iotEdgeGatewayHostname))
+            {
+                throw new ArgumentException("IoT Edge gateway hostname cannot be null or empty.");
+            }
+
+            if (string.IsNullOrWhiteSpace(deviceIdentityCertPath) || !File.Exists(deviceIdentityCertPath))
+            {
+                throw new ArgumentException($"Downstram device identity certificate path is invalid");
+            }
+
+            if (string.IsNullOrWhiteSpace(deviceIdentityPrivateKeyPath) || !File.Exists(deviceIdentityPrivateKeyPath))
+            {
+                throw new ArgumentException($"Downstram device identity private key path is invalid");
+            }
+
+            InstallCACert();
+
+            try
+            {
+                string messageCountEnv = Environment.GetEnvironmentVariable("MESSAGE_COUNT");
```

Comment:
Use Configuration (same as SimulatedTemperatureSensor) and add Env vatiables. You can use configuration.GetValue<int>("MESSAGE_COUNT", 10),